### PR TITLE
[DEV]Add task order field for project.task

### PR DESCRIPTION
**Pull request by jananjoy**

### CODE REVIEW

File: project_event/models/project_task.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
 
 from odoo import api, fields, models
+from datetime import datetime, time, timedelta
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @jananjoy time and timedelta imports are not used

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -68,6 +69,10 @@ class Task(models.Model):
         store=True,
         readonly=True,
     )
+    task_order = fields.Integer(
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:72]: @jananjoy this field should be readonly and stored field

File: project_event/models/project_task.py [hunk 3]
```diff
@@ -94,3 +99,13 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
             domain = ['|', ('name', operator, name),
                       ('code', operator, name)]
         return super(Task, self).search(domain + args, limit=limit).name_get()
+
+    @api.one
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @jananjoy @api.depends(start_date,parent_id.start_date)

File: project_event/models/project_task.py [hunk 4]
```diff
@@ -94,3 +99,13 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
...
+    def _compute_order_task(self):
+        fmt = '%Y-%m-%d %H:%M:%S'
+        if self.parent_id != False:
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @jananjoy simplify this expression

File: project_event/models/project_task.py [hunk 5]
```diff
@@ -94,3 +99,13 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
...
+            if self.parent_id.date_start != False:
+                activity_start_date = self.parent_id.date_start
+                if self.activity_task_type  == 'task':
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @jananjoy better to start with this condition and delete the space before ==

File: project_event/models/project_task.py [hunk 6]
```diff
@@ -94,3 +99,13 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
...
+                    time_difference = datetime.strptime(self.date_start, fmt) - datetime.strptime(activity_start_date, fmt)
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @jananjoy PEP8 line too long (123>80 characters)

File: project_event/views/project_task_view.xml [hunk 7]
```diff
@@ -121,6 +121,7 @@
                 <field name="parent_id" string="Activity"/>
                 <field name="date_start"/>
                 <field name="date_end"/>
+                <field name="task_order"/>
```
Comment (rimbendhaou) [inline project_event/views/project_task_view.xml]: @jananjoy this field will be invisible

File: project_event/models/project_task.py (hunk 4 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou Need for the precise explanation. Is it on format expression or condition checking.

Comment (rimbendhaou) [reply project_event/models/project_task.py]: @jananjoy if self.parent_id

File: project_event/models/project_task.py [hunk 8]
```diff
@@ -94,3 +101,17 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
             domain = ['|', ('name', operator, name),
                       ('code', operator, name)]
         return super(Task, self).search(domain + args, limit=limit).name_get()
+
+    @api.one
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @jananjoy delete @api.one

File: project_event/models/project_task.py [hunk 9]
```diff
@@ -94,3 +101,17 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
...
+    @api.depends('date_start', 'parent_id.date_start')
+    def _compute_order_task(self):
+        self.ensure_one()
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @jananjoy for task in self:

Comment (rimbendhaou) [inline project_event/models/project_task.py]: @jananjoy delete self.ensure_one

**Review by rimbendhaou — APPROVED** [review]
LGTM

**Pull request merged**